### ISSUE TITLE
API change from Microsoft Breaks existing method

### ISSUE THREAD

**Issue by SushmaShetty** [MicrosoftEdge/WebView2Feedback#3399]
**New Method**

// Microsoft.Web.WebView2.Core, Version=1.0.1587.40, Culture=neutral, PublicKeyToken=2a8ab48044d2601e

// Microsoft.Web.WebView2.Core.CoreWebView2EnvironmentOptions

public CoreWebView2EnvironmentOptions(string additionalBrowserArguments = null, string language = null, string targetCompatibleBrowserVersion = null, bool allowSingleSignOnUsingOSPrimaryAccount = false, List<CoreWebView2CustomSchemeRegistration> customSchemeRegistrations = null)

{}

**OLD:**

// Microsoft.Web.WebView2.Core, Version=1.0.1418.22, Culture=neutral, PublicKeyToken=2a8ab48044d2601e

// Microsoft.Web.WebView2.Core.CoreWebView2EnvironmentOptions

public CoreWebView2EnvironmentOptions(string additionalBrowserArguments = null, string language = null, string targetCompatibleBrowserVersion = null, bool allowSingleSignOnUsingOSPrimaryAccount = false)

{}

Please let me know if this has be adrressed.

**Comment by novac42** [comment]
Thanks for reaching out. I've assigned this to a dev that can help follow up on this.

**Comment by LiangTheDev** [comment]
@SushmaShetty could you share details on how the app uses the WebView2 SDK/API and version of WebView2 Runtime?

If the app uses WebView2 SDK, everything should work.

I've seen some app provide their own implementation of environment option object, and as long as their object response to QueryInterface correctly, the app should also work with newer WebView2 Runtime.
If the app's environment option handles QueryInterface incorrectly, like always return itself for whatever interface even when the interface is not implemented, then crash will happen when trying to call the non existing methods of the not implemented interface.

**Comment by SushmaShetty** [comment]
yeah Below code  which is complied using : 1.0.148.22 and implementation done
  CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions((null), (null), (null), false)
            {
                AdditionalBrowserArguments = "test"
            };
            CoreWebView2CreationProperties CreationProperties = new CoreWebView2CreationProperties();
            CreationProperties.BrowserExecutableFolder = Common.RuntimePath;
            CoreWebView2Environment.CreateAsync(CreationProperties.BrowserExecutableFolder, CreationProperties.UserDataFolder, options);
            ShuiwebView.Source = new Uri("https://www.microsoft.com/en-in");

Now at run time we upgrade to Latest version :1.01587.40 application crashes.

**Comment by LiangTheDev** [comment]
It is by design. When you adopt a new SDK, you have to recompile the app. The app is built against a certain assembly reference when it is compiled, and the assemblies in the SDK has those APIs. In a newer version of SDK, the API signature could change and the app built against the old version would not match the assemblies in the newer SDK.

To clarify, the Edge WebView2 Runtime is a separate product that installed on the machine, and that is automatically updated and compatible with apps build with old nuget package.
The things in nuget package is SDK, which is used to build the app. After adopting a new SDK, you should rebuild the app.

**Comment by SushmaShetty** [comment]
In our case Product contains of multiple components, each developed by different teams and also companies.
Multiple components are using the WebView2 controls and are plugged into a single process in the final product.
The new default arguments you have introduced are now forcing all components to use the same WebView2 WPF control version, which is not really feasible at our site to coordinate this (as every component has its own release cycle).

Therefore overloads instead of default arguments would help in this scenario . To be runtime compatible and not just compile time compatible.
